### PR TITLE
Reads message_type from ClientMessage.

### PR DESCRIPTION
**Pull request by PLehmColCom**
See: https://tronche.com/gui/x/xlib/events/client-communication/client-message.html

### CODE REVIEW

**Comment by sidorares** [comment]
Thanks! Any chance to add test?

**Comment by PLehmColCom** [comment]
I just made a little fix so the Datatype returned by message_type is Number and not Array. Also I tried to engineer a proper test. 
Since I don't have much experience with the X11 protocol yet maybe you can look over it again. Just to be sure. ;)

File: test/client-message.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                should.not.exist(err);
+                ATOM['TEST_ATOM_2'] = atom;
+                done();
+              });
+            });
+      });
+
+      client.on('error', function (err) {
+          console.error('Error : ', err);
+      });
```
Comment (santigimeno) [inline test/client-message.js]: `client.on('error', done);` ?

**Comment by santigimeno** [comment]
I think it looks good. Thanks!

**Comment by PLehmColCom** [comment]
Got
```
  client.on('error', function (err) {
    console.error('Error : ', err);
  });
```
from the ``configure-request`` and ``configure-window`` tests. But changed it to ``client.on('error', done);`` now. No problem. This library is great! :+1:

**Comment by sidorares** [comment]
lgtm! Thanks a lot

**Pull request merged**